tests/jenkins/ansible-advanced-three-tier-app.groovy

```diff
@@ -0,0 +1,278 @@
+            description: 'Catalog item',
+            name: 'region',
+        )
+    }
+
+    stages {
+        stage('order from CF') {
+            environment {
+                uri = "${cf_uri}"
+                credentials = credentials("${opentlc_creds}")
+                DEBUG = 'true'
+            }
+            /* This step use the order_svc_guid.sh script to order
+             a service from CloudForms */
+            steps {
+                git url: 'https://github.com/fridim/cloudforms-oob'
+
+                script {
+                    def catalog = params.catalog_item.split(' / ')[0].trim()
+                    def item = params.catalog_item.split(' / ')[1].trim()
+                    def region = params.region.trim()
+                    def cfparams = [
+                        'expiration=7',
+                        'runtime=8',
+                        "region=${region}",
+                    ].join(',').trim()
+
+                    echo "'${catalog}' '${item}'"
+                    guid = sh(
+                        returnStdout: true,
+                        script: """
+                          ./opentlc/order_svc_guid.sh \
+                          -c '${catalog}' \
+                          -i '${item}' \
+                          -G '${cf_group}' \
+                          -d '${cfparams}' \
+                        """
+                    ).trim()
+
+                    echo "GUID is '${guid}'"
+                }
+            }
+        }
+
+        stage('Wait for first email') {
+            environment {
+                credentials=credentials("${imap_creds}")
+            }
+            steps {
+                git url: 'https://github.com/sborenst/ansible_agnostic_deployer',
+                    branch: 'development'
+
+
+                sh """./tests/jenkins/downstream/poll_email.py \
+                    --server '${imap_server}' \
+                    --guid ${guid} \
+                    --timeout 20 \
+                    --filter 'has started'"""
+            }
+        }
+
+        stage('Wait for last email and parse OpenShift and App location') {
+            environment {
+                credentials=credentials("${imap_creds}")
+            }
+            steps {
+                git url: 'https://github.com/sborenst/ansible_agnostic_deployer',
+                    branch: 'development'
+
+                script {
+                    email = sh(
+                        returnStdout: true,
+                        script: """
+                          ./tests/jenkins/downstream/poll_email.py \
+                          --server '${imap_server}' \
+                          --guid ${guid} \
+                          --timeout 40 \
+                          --filter 'has completed'
+                        """
+                    ).trim()
+
+                    try {
+                        def m = email =~ /ssh -i ~/.ssh/your_private_key_name ([^ \n]+)/
```

Comment:
Is this regex working ?
You don't have to escape the `/` ?

```groovy
                        def m = email =~ /ssh -i ~\/.ssh\/your_private_key_name ([^ \n]+)/
```

Reply:
fixed.